_episodes/l2_02_basic_unit_testing.md

```diff
@@ -0,0 +1,285 @@
+- `Test logic and assertions`: This is the body of the test case where you write the actual test code, including any necessary assertions to verify the expected behaviour. The section `Test Assertions` describes more about various assertions available in GoogleTest.
+
+### 3.3. Initialise GoogleTest framework in your main
+
+When you install GoogleTest you would get two libraries namely `libgtest.a` and `libgtest_main.a` (The extension `.a` means it is a static library and is applicable for Linux based system) . The first one i.e. `libgtest.a` is the library which provides all the necessary testing features such as assertions, test discovery, collection of results etc. The second library i.e. `libgtest_main.a` provides a main function so that you do not need to write your own main function for testing. For more details, refer to https://stackoverflow.com/questions/6457856/whats-the-difference-between-gtest-lib-and-gtest-main-lib.  
+
+For this sub-section, we are assuming that you are writing your own main function while the next section describes how to  run your tests without your own main function by linking to the main function provided in `lgtest_main`. You would need to include the following lines in your main function as shown below.
+
+```cpp
+int main(int argc, char **argv)
+{
+    // Initialise GoogleTest Framework
+    testing::InitGoogleTest(&argc, argv);
+    
+    // Instruction to Run the test cases.
+    return RUN_ALL_TESTS();
+}
+```
+
+The `RUN_ALL_TESTS()` is typically called from `main` and when it is invoked, it scans the program for all the test cases and  test methods defined using the `test` macro. It then executes each test case and captures the results of each individual test method within the test case. After running all the tests, it provides a summary of the test results, including the number of tests run, passed, and failed.
+
+### 3.4. Compile and run your program
+
+In order to compile your program and link with the required libraries, you can use the following instruction as a template. Please remember to modify the paths according to your system and location of the files.
+
+```bash
+# For compiling and linking, please use the instruction below
+$ g++ your_code.cpp -I path_to_your_gtest.h -L path_to_your_lgtest.a -lgtest -lpthread -o your_executable_name
+ 
+# Run your code  
+$ ./your_executable_name
+```
+
+In the above cell, `lpthread` refers to `pthread` library which is an acronym for "POSIX Threads". It is a library in C and C++ that provides an interface for creating and managing threads in a multi-threaded program. It is based on the POSIX (Portable Operating System Interface) standard for thread management. 
+
+Although, we are not using any multi-threaded feature explicity in this course, we still need to link the progam with this library. This is because some of the function described in `lgtest` depends on `lpthread` and without it, linker will give an error. 
```

Comment:
```suggestion
Although, we are not using any multi-threaded feature explicity in this course, we still need to link the progam with this library. This is because some of the function described in `lgtest` depends on `lpthread` and without it the linker will give an error. 
```